### PR TITLE
fix: Make column directive reactive for input props

### PR DESCRIPTION
**Pull request by Dagdelo**
The current implementation lacks reactivity for the input props. Now, when a prop changes the directive will reflect these changes. This feature is required when the application needs to dynamically change the columns' disposition.

Closes carbon-design-system/carbon-components-angular#2385

### CODE REVIEW

**Comment by Dagdelo** [comment]
recheck

**Comment by Dagdelo** [comment]
recheck

**Comment by Akshat55** [comment]
@Dagdelo Going to review this sometime next week. There was some changes made in next branch that we can use here. We _**might**_ be able to use observable here.

File: src/grid/grid.directive.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Directive, HostBinding, Input, OnInit } from "@angular/core";
+import { Directive, HostBinding, Input, OnInit, OnChanges, SimpleChanges } from "@angular/core";
```
Comment (Akshat55) [inline src/grid/grid.directive.ts:1]:
```suggestion
import { 
	Directive,
	HostBinding,
	Input,
	OnChanges,
	SimpleChanges
} from "@angular/core";
```

File: src/grid/grid.directive.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class RowDirective {
 @Directive({
 	selector: "[ibmCol]"
 })
-export class ColumnDirective implements OnInit {
+export class ColumnDirective implements OnInit, OnChanges {
```
Comment (Akshat55) [inline src/grid/grid.directive.ts:40]:
```suggestion
export class ColumnDirective implements OnChanges {
```

File: src/grid/grid.directive.ts [hunk 3]
```diff
@@ -78,7 +78,15 @@ export class ColumnDirective implements OnInit {
...
 		}
 	}
+
+	ngOnInit() {
+		this.setUp();
+	}
+
+	ngOnChanges(changes: SimpleChanges): void {
+    	this.setUp();
+  	}
```
Comment (Akshat55) [inline src/grid/grid.directive.ts:91]:
```suggestion
```

File: src/grid/grid.directive.ts (hunk 2 above)
Comment (Akshat55) [reply src/grid/grid.directive.ts:40]: We can just remove OnInit and add the logic inside OnChanges

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/grid/grid.directive.ts [hunk 4]
```diff
@@ -1,4 +1,10 @@
-import { Directive, HostBinding, Input, OnInit } from "@angular/core";
+import { 
+	Directive,
+	HostBinding,
+	Input,
+	OnChanges,
+	SimpleChanges
```
Comment (Akshat55) [inline src/grid/grid.directive.ts:6]:
```suggestion
	OnChanges
```

File: src/grid/grid.directive.ts [hunk 5]
```diff
@@ -55,8 +61,10 @@ export class ColumnDirective implements OnInit {
 		this._columnClasses = classes.split(" ");
 	}
 
-	ngOnInit() {
+	ngOnChanges() {
 		try {
+			// Reset classes so we don't apply classes for the same breakpoint multiple times
+			this.columnClasses = "";
```
Comment (Akshat55) [inline src/grid/grid.directive.ts:67]:
```suggestion
			this._columnClasses = "";
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**